Tasks/Maven/pmdformaven.ts

```diff
@@ -10,25 +10,27 @@ import trm = require('vsts-task-lib/toolrunner');
 
 import ar = require('./analysisresult');
 
+export const toolName:string = 'PMD';
+
 // Adds PMD goals, if selected by the user
 export function applyPmdArgs(mvnRun: trm.ToolRunner):void {
-    mvnRun.arg(['jxr:jxr', 'pmd:pmd']); // JXR source code cross-referencing is a required pre-goal for PMD
+    mvnRun.arg(['pmd:pmd', '-DlinkXRef=false']); // Turn off cross-referencing to reduce Maven error output
 }
 
 // Extract analysis results from PMD output file.
-// Takes the current working directory and returns an AnalysisResult data class.
+// Takes the working directory (should contain pom.xml and target/) and returns an AnalysisResult data class.
 // Task fails if the HTML or XML outputs were not found.
 // @param sourcesDirectory - The absolute location of the root source directory.
-export function processPmdOutput(sourcesDirectory:string) : ar.AnalysisResult {
+export function processPmdOutput(rootDir:string) : ar.AnalysisResult {
     var result:ar.AnalysisResult = new ar.AnalysisResult();
-    result.toolName = 'PMD';
+    result.toolName = toolName;
 
-    var pmdXmlFilePath = path.join(sourcesDirectory, '/target/pmd.xml');
+    var pmdXmlFilePath = path.join(rootDir, 'target', 'pmd.xml');
     result = processPmdXml(result, pmdXmlFilePath);
 
     // if there are no violations, there will be no HTML report
     if (result.totalViolations > 0) {
-        var pmdHtmlFilePath = path.join(sourcesDirectory, '/target/site/pmd.html');
+        var pmdHtmlFilePath = path.join(rootDir, 'target', 'site', 'pmd.html');
```

Comment:
This appears to be the only point that the .xml and html files are treated differently depending on which file it is i.e. the _AnalysisResult_ should just have a list of "FilesToUpload" which both can be added to.